### PR TITLE
chore: Update pre-commit checks to ensure common standards are followed

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
- Update pre-commit checks to ensure common standards are followed; see [Rules](https://github.com/terraform-linters/tflint/tree/master/docs/rules) for full list of available rules that tflint offers

### Motivation
- Ensure that we provide all the relevant configurations and only those that are required/used
- Ensure module follows standards and conventions adopted by wider community

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by bryantbiggs** [comment]
Opening early to show the output caught by these changes in CI

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
+# TODO - different name since if its not going to run on PRs
```
Comment (bryantbiggs) [inline .github/workflows/pr-test.yml:1]: This workflow requires access to secrets for `ROLE_TO_ASSUME` which open source contributors outside the org will not have access to which results in the error

```
Error: Credentials could not be loaded, please check your action inputs: Could not load credentials from any providers
```

I think most of what happens in this workflow is duplicated in https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/main/.github/workflows/pre-commit.yaml, we could probably remove this and rely on the other tests we have in place. TBD

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -1,97 +1,77 @@
 name: pre-commit
 
 on:
-  push:
```
Comment (bryantbiggs) [inline .github/workflows/pre-commit.yaml:4]: This is better to show users on pull-requests and doesn't require any credential access so flipping to run on PRs

Comment (bryantbiggs) [reply .github/workflows/pre-commit.yaml:4]: Most of the other changes are just updates from the original workflow I created which now uses composite workflows

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -10,9 +10,26 @@ repos:
       - id: detect-aws-credentials
         args: ['--allow-missing-credentials']
   - repo: https://github.com/antonbabenko/pre-commit-terraform
-    rev: v1.53.0
+    rev: v1.62.0
```
Comment (bryantbiggs) [inline .pre-commit-config.yaml:13]: For now we should avoid going above v1.62.0 until https://github.com/antonbabenko/pre-commit-terraform/issues/337 is fixed without work arounds

**Review by vara-bonthu — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml (hunk 3 above)
Comment (bryantbiggs) [reply .pre-commit-config.yaml:13]: Fixed in v1.64.1 https://github.com/antonbabenko/pre-commit-terraform/pull/339

**Comment by bryantbiggs** [comment]
Closing for now to keep focus on launch activities - will re-open later

**Comment by bryantbiggs** [comment]
Closing for now to keep focus on launch activities - will re-open later

**Pull request closed without merging**